src/main/java/the_warlord/powers/BoobyTrapPower.java

```diff
@@ -0,0 +1,54 @@
+package the_warlord.powers;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
+import com.megacrit.cardcrawl.actions.common.DamageAction;
+import com.megacrit.cardcrawl.actions.common.RemoveSpecificPowerAction;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.localization.PowerStrings;
+import com.megacrit.cardcrawl.powers.AbstractPower;
+import the_warlord.WarlordMod;
+import the_warlord.cards.warlord.BoobyTrap;
+
+public class BoobyTrapPower extends AbstractPower {
```

Comment:
I would prefer to see your powers as extensions of CustomWarlordModPower, so that we have a generic parent that we can add functionality to. It's also much fewer lines of code and much less boilerplate code.